x/payments/internal/types/params.go

```diff
@@ -133,18 +97,218 @@ func (p Params) String() string {
 	)
 }
 
+func validateFactor(i interface{}) error {
+	v, ok := i.(sdk.Dec)
+
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	if v.IsNil() {
+		return fmt.Errorf("ixo factor must be not nil")
+	}
+	if v.IsNegative() {
+		return fmt.Errorf("ixo factor must be positive: %s", v)
+	}
+	if v.GT(sdk.OneDec()) {
+		return fmt.Errorf("ixo factor too large: %s", v)
+	}
+
+	return nil
+}
+
+func validateFeeAmount(i interface{}) error {
```

Comment:
Function name should match the param we are validating. In this case it is `InitiationFeeAmount` and so the function should be `validateInitiationFeeAmount`

Reply:
ok